lib/base/process.cpp

```diff
@@ -109,11 +109,11 @@ static Value ProcessSpawnImpl(struct msghdr *msgh, const Dictionary::Ptr& reques
 	int j = 0;
 
 	for (int i = 0; i < envc; i++) {
-		if (strncmp(environ[i], lcnumeric, strlen(lcnumeric)) == 0) {
+		if (strcmp(environ[i], lcnumeric) == 0) {
 			continue;
 		}
 
-		if (strncmp(environ[i], notifySocket, strlen(notifySocket)) == 0) {
+		if (strcmp(environ[i], notifySocket) == 0) {
 			continue;
 		}
```

Comment:
That's actually not just a redundant overflow protection but is actually implementing a prefix check here. But that's totally not obvious here, I also got this wrong first. So changing the code to make this more obvious would actually be a quality improvement.

Reply:
## TODO

* [ ] #9719
* [ ] rebase